pkg/operator/clustermembercontroller/clustermembercontroller.go

```diff
@@ -573,3 +593,34 @@ func (c *ClusterMemberController) getResyncName(pods *corev1.PodList) (string, e
 	}
 	return "", nil
 }
+
+func isClusterEtcdOperatorReady(originalSpec *operatorv1.OperatorSpec) bool {
+	spec := originalSpec.DeepCopy()
+
+	// don't worry about errors.  If we can't decode, we'll simply stomp over the field.
+	existingConfig := map[string]interface{}{}
+	if err := json.NewDecoder(bytes.NewBuffer(spec.ObservedConfig.Raw)).Decode(&existingConfig); err != nil {
+		klog.V(4).Infof("decode of existing config failed with error: %v", err)
+	}
+
+	membersPath := []string{"cluster", "members"}
+	membersPendingPath := []string{"cluster", "pending"}
+
+	members, _, err := unstructured.NestedSlice(existingConfig, membersPath...)
+	if err != nil {
+		klog.Errorf("error reading members: %#v", err)
+		return false
+	}
+	klog.Infof("observed members: %#v", members)
+	pendingMembers, _, err := unstructured.NestedSlice(existingConfig, membersPendingPath...)
+	if err != nil {
+		klog.Errorf("error reading pending members: %#v", err)
+		return false
+	}
+	if len(pendingMembers) > 0 {
```

Comment:
Question: what if we are missing a node somehow? Don't we want to verify N are healthy? Just want to make sure this is enough to say we have succeeded.

Reply:
I know we have this issue of N state do we need to deal with that now?

Reply:
I would like to deal with the node issue later on with CI in place honestly, what do you think? 

even if we handle it right now how do we test it? it would open up many issues IMO, fight for another day :)

Reply:
I think we can get this N from installer.

N := installConfig.Config.ControlPlane.Replicas+1

Reply:
probably should be a flag into render and added to config

Reply:
I am ok with pushing this a little bit but it should be a p0 effort as we need this state for Verizon I bet.

Reply:
https://github.com/openshift/installer/blob/79dae77251693ad5eac0f3619efa983e589121a1/pkg/asset/manifests/operators.go#L169

Reply:
I agree with p0 effort

I think we should save it in `etcds.Spec` so that we survive across restarts it. Would it make sense to have the render function spit out an etcd cr yaml with managed/unmanaged and appropriate number of replicas? 

So all we have to do in the operator is read it from the Spec and compare...

Reply:
yes